File: platform-mobile/src/main/java/ru/pocketbyte/locolaser/platform/mobile/resource/AbsIosStringsResources.java
package ru.pocketbyte.locolaser.platform.mobile.resource;

import ru.pocketbyte.locolaser.resource.AbsPlatformResources;

import java.io.File;

public abstract class AbsIosStringsResources extends AbsPlatformResources {

    public static final String RES_FILE_EXTENSION = ".strings";

    public AbsIosStringsResources(File resourcesDir, String name) {
        super(resourcesDir, name);
    }

    protected File getFileForLocale(String locale) {
        File localeFolder = new File(getDirectory(), getLocaleDirName(locale));
        localeFolder.mkdirs();
        return new File(localeFolder, getName() + RES_FILE_EXTENSION);
    }

    protected String getLocaleDirName(String locale) {
        StringBuilder builder = new StringBuilder();
        if (BASE_LOCALE.equals(locale))
            builder.append("Base");
        else
            builder.append(locale);
        builder.append(".lproj");
        return builder.toString();
    }
}

File: platform-mobile/src/main/java/ru/pocketbyte/locolaser/platform/mobile/resource/AbsIosStringsResources.kt
package ru.pocketbyte.locolaser.platform.mobile.resource

import ru.pocketbyte.locolaser.resource.AbsPlatformResources
import ru.pocketbyte.locolaser.resource.PlatformResources

import java.io.File

abstract class AbsIosStringsResources(
        resourcesDir: File,
        name: String
) : AbsPlatformResources(resourcesDir, name) {

    companion object {
        const val RES_FILE_EXTENSION = ".strings"
    }

    protected fun getFileForLocale(locale: String): File {
        val localeFolder = File(directory, getLocaleDirName(locale))
        localeFolder.mkdirs()
        return File(localeFolder, name + RES_FILE_EXTENSION)
    }

    protected fun getLocaleDirName(locale: String): String {
        val builder = StringBuilder()
        if (PlatformResources.BASE_LOCALE == locale)
            builder.append("Base")
        else
            builder.append(locale)
        builder.append(".lproj")
        return builder.toString()
    }
}